packages/runtime/container-runtime/src/opLifecycle/opCompressor.ts

```diff
@@ -7,53 +7,58 @@ import { ITelemetryLogger } from "@fluidframework/common-definitions";
 import { IsoBuffer } from "@fluidframework/common-utils";
 import { ChildLogger } from "@fluidframework/telemetry-utils";
 import { compress } from "lz4js";
-import { BatchMessage } from "./batchManager";
-import { CompressionAlgorithms, ContainerRuntimeMessage } from "./containerRuntime";
+import { CompressionAlgorithms, ContainerRuntimeMessage } from "../containerRuntime";
+import { IBatchProcessor, IBatch, BatchMessage } from "./definitions";
 
 /**
  * Compresses batches of ops. It generates a single compressed op that contains
  * the contents of each op in the batch. It then submits empty ops for each original
  * op to reserve sequence numbers.
  */
-export class OpCompressor {
+export class OpCompressor implements IBatchProcessor {
     private readonly logger;
     private compressedBatchCount = 0;
 
     constructor(logger: ITelemetryLogger) {
         this.logger = ChildLogger.create(logger, "OpCompressor");
     }
 
-    public compressBatch(batch: BatchMessage[], originalLength: number): BatchMessage[] {
-        const batchToSend: BatchMessage[] = [];
+    public processOutgoing(batch: IBatch): IBatch {
+        const messages: BatchMessage[] = [];
         this.compressedBatchCount++;
-        const batchedContents: ContainerRuntimeMessage[] = [];
-        for (const message of batch) {
-            batchedContents.push(message.deserializedContent);
+        const contentToCompress: ContainerRuntimeMessage[] = [];
+        for (const message of batch.content) {
+            contentToCompress.push(message.deserializedContent);
         }
 
         const compressionStart = Date.now();
-        const contentsAsBuffer = new TextEncoder().encode(JSON.stringify(batchedContents));
+        const contentsAsBuffer = new TextEncoder().encode(JSON.stringify(contentToCompress));
         const compressedContents = compress(contentsAsBuffer);
         const compressedContent = IsoBuffer.from(compressedContents).toString("base64");
         const duration = Date.now() - compressionStart;
 
-        if (originalLength > 200000 || this.compressedBatchCount % 100) {
+        if (batch.contentSizeInBytes > 200000 || this.compressedBatchCount % 100) {
```

Comment:
We should probably either remove this limit / sampling or reduce it.
As of now, the plan of record is to enable compression only for large payloads. I.e. we will get here very infrequently. With the way we filter / sample, we will have almost no data in telemetry to work with